File: api/tenant/v1/marketplace.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

type MarketplaceListReq struct {
	g.Meta   `path:"/marketplace/models" method:"get" tags:"公开-模型广场" summary:"获取模型广场列表" group:"public" middleware:"-"`
	Keyword  string `json:"keyword" dc:"搜索关键词"`
	Category string `json:"category" dc:"模型类别筛选（chat/embedding/image/audio等）"`
	Vendor   string `json:"vendor" dc:"研发厂商筛选（openai/anthropic/alibaba等），空=不过滤"`
	Page     int    `json:"page" d:"1" dc:"页码"`
	PageSize int    `json:"page_size" d:"20" v:"between:1,100" dc:"每页数量"`
}

type MarketplaceListRes struct {
	List     []MarketplaceModelItem `json:"list"`
	Total    int                    `json:"total"`
	Page     int                    `json:"page"`
	PageSize int                    `json:"page_size"`
	// Vendors 当前筛选条件（关键词/分类，忽略厂商筛选本身）下有模型的厂商去重列表，
	// 供前端只渲染实际有模型的厂商筛选项；顺序不保证，前端按本地字典序展示
	Vendors []string `json:"vendors"`
}

type MarketplaceModelItem struct {
	ModelId            string             `json:"model_id"`                    // 模型标识
	ModelName          string             `json:"model_name"`                  // 模型显示名称
	Category           string             `json:"category"`                    // 分类（chat/embedding/image等）
	Vendor             string             `json:"vendor"`                      // 研发厂商（openai/anthropic/alibaba等，空=未分类）
	Description        string             `json:"description"`                 // 描述
	MaxContextTokens   int                `json:"max_context_tokens"`          // 最大上下文 tokens
	MaxOutputTokens    int                `json:"max_output_tokens"`           // 最大输出 tokens
	BillingMode        string             `json:"billing_mode"`                // 计费模式：token/per_request/tiered/per_second
	InputPrice         float64            `json:"input_price"`                 // 输入价格（每百万 token，USD；tiered=首档）
	OutputPrice        float64            `json:"output_price"`                // 输出价格（每百万 token，USD；tiered=首档）
	PerRequestPrice    float64            `json:"per_request_price"`           // 按次单价（USD/次，仅 per_request）
	PerSecondPrices    map[string]float64 `json:"per_second_prices,omitempty"` // 按秒单价矩阵（规格→USD/秒，仅 per_second），"*"为兜底价
	CacheReadPrice     float64            `json:"cache_read_price"`            // 缓存读取价格（每百万 token，USD）
	CacheCreationPrice float64            `json:"cache_creation_price"`        // 缓存创建价格（每百万 token，USD）
	DiscountLabel      string             `json:"discount_label"`              // 折扣标签（营销展示，空=不展示）
	PriceChangeNote    string             `json:"price_change_note"`           // 价格调整说明（对外提示，空=不展示）
	TimePrices         []TimePriceItem    `json:"time_prices"`                 // 时段价目（平台基础价 × 时段乘数，无配置=空）
	Tags               []string           `json:"tags"`                        // 标签
	Capabilities       g.Map              `json:"capabilities"`                // 能力标签
}

type MarketplaceDetailReq struct {
	g.Meta  `path:"/marketplace/models/{model_id}" method:"get" tags:"公开-模型广场" summary:"获取模型详情" group:"public" middleware:"-"`
	ModelId string `json:"model_id" in:"path" v:"required" dc:"模型标识"`
}

type MarketplaceDetailRes struct {
	MarketplaceModelItem
}

File: web/tenant/src/api/marketplace.ts
import request from '@/utils/request'

export interface TimePriceItem {
  name: string
  days?: number[] | null
  start_time?: string
  end_time?: string
  valid_from?: string
  valid_to?: string
  multiplier: number
  input_price?: number | null
  output_price?: number | null
  per_request_price?: number | null
}

export interface MarketplaceModel {
  model_id: string
  model_name: string
  category: string
  /** 研发厂商枚举（openai/anthropic/alibaba 等，空串=未分类） */
  vendor?: string | null
  description: string
  max_context_tokens: number
  max_output_tokens: number
  billing_mode?: string
  input_price: number
  output_price: number
  per_request_price?: number
  cache_read_price?: number
  cache_creation_price?: number
  discount_label?: string | null
  price_change_note?: string | null
  time_prices?: TimePriceItem[] | null
  tags: string[]
  capabilities: Record<string, any>
}

export interface MarketplaceListParams {
  keyword?: string
  category?: string
  vendor?: string
  page: number
  page_size: number
}

export interface MarketplaceListResponse {
  list: MarketplaceModel[]
  total: number
  page: number
  page_size: number
  /** 当前条件下有模型的厂商去重列表（前端只渲染这些厂商筛选项） */
  vendors?: string[] | null
}

export const getMarketplaceModels = (params: MarketplaceListParams) => {
  return request.get<{ data: MarketplaceListResponse }>('/tenant/marketplace/models', { params }).then(res => res.data.data)
}

export const getMarketplaceModelDetail = (modelId: string) => {
  return request.get<{ data: MarketplaceModel }>(`/tenant/marketplace/models/${modelId}`).then(res => res.data.data)
}
